src/components/BackToTop.vue

```diff
@@ -13,37 +13,29 @@
   </div>
 </template>
 
-<script>
-
+<script setup>
 import { ref } from 'vue';
 
+const show = ref(false)
+
+let props = defineProps({targetId: {
+  type: String,
+  default: '',
+  required: false
+}})
 
-export default {
-  name:'BackToTop',
-  props: {
-    targetId: {
-      type: String,
-      default: '',
-      required: false
-    },
-  },
-  data: () => ({
-    show : ref(false),
-  }),
-  methods: {
-    onScroll(e) {
-      const scrollTopValue = this.targetId ? e.target.scrollTop : window.scrollY ;
-      this.show = scrollTopValue > 200;
-    },
-    onClick() {
-      const targetSchroll = this.targetId ? document.getElementById('rulesDialog') : window ;
-      targetSchroll.scrollTo({
-        top: 0,
-        behavior: 'smooth',
-      });
-    }
-  }
-};
+function onScroll(e) {
+  const scrollTopValue = props.targetId ? e.target.scrollTop : window.scrollY ;
```

Comment:
We generally have omitted whitespace before semicolons. Not sure if there is a linter rule we could apply to more automate making this consistent.
```suggestion
  const scrollTopValue = props.targetId ? e.target.scrollTop : window.scrollY;
```